### ISSUE TITLE
Tuple values in tail are not type checked

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#464]
**Describe the bug**
Pyright only checks the first item in a tuple for type correctness and homogenously-typed tuples that contain heterogenous values do not produce errors.  This does not only affect tuples annotated with `Tuple` but all iterable generics that apply to tuples.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That disparate types in tuples would be rejected.

**Screenshots or Code**
```
from typing import Tuple


def bar(values: Tuple[str, ...]): ...


bar(('', False))        # this passes
bar((False, ''))        # this one doesn't
```

**VS Code extension or command-line**
extension, 1.1.16

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This fix is included in Pyright v1.1.17, which I just published.
